### PR TITLE
Fix elasticsearch index

### LINKED ISSUE #2971: Bug in elasticsearch index of metrics server
## Describe the bug

[Here](https://github.com/SeldonIO/seldon-core/blob/63413ad30fd9d454ddee8974a4d35b49c28d86bf/components/alibi-detect-server/adserver/cm_model.py#L140), the `elasticsearch_index`  appears to be constructed in the wrong way: 

This:
```python
# Currently only supports SELDON inference type (not kfserving)
                    elasticsearch_index = f"inference-log-{seldon_namespace}-seldon-{SELDON_DEPLOYMENT_ID}-{SELDON_PREDICTOR_ID}"
```
should rather be:
```python
# Currently only supports SELDON inference type (not kfserving)
                    elasticsearch_index = f"inference-log-seldon-{seldon_namespace}-{SELDON_DEPLOYMENT_ID}-{SELDON_PREDICTOR_ID}"
```

**Reason:**
When deploying a model to the namespace `production`, the elasticsearch index is named as follows:
`'inference-log-seldon-production-multiclass-model-default'`

## To reproduce

The easiest way to reproduce this bug is to follow [this](https://github.com/SeldonIO/seldon-core/blob/master/examples/feedback/feedback-metrics-server/README.ipynb) notebook from the seldon examples but instead of choosing the namespace `seldon` choose a different one so that the error can manifest:

```
kubectl create namespace seldon_debug || echo "namespace already created"
kubectl config set-context $(kubectl config current-context) --namespace=seldon_debug
```

## Expected behaviour
When deploying a metrics server that get's triggered when feedback is sent, it is supposed to lookup the respective document in elasticsearch. This, however only works by chance if the namespace is `seldon` as then the bug described above does not manifest.

#### Kubernetes version:
 Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.14-gke.1600", GitCommit:"7c407f5cc8632f9af5a2657f220963aa7f1c46e7", GitTreeState:"clean", BuildDate:"2020-12-07T09:22:27Z", GoVersion:"go1.13.15b4", Compiler:"gc", Platform:"linux/amd64"}

#### Seldon Images:
          value: docker.io/seldonio/engine:1.6.0
          value: docker.io/seldonio/seldon-core-executor:1.6.0
        image: docker.io/seldonio/seldon-core-operator:1.6.0

## Model Details 
Logs of the metrics server:

```
[I 210216 16:33:11 __main__:111] Extra args: {}
[I 210216 16:33:11 server:122] Registering model:multiclassserver
[I 210216 16:33:11 server:113] Listening on port 8080
[I 210216 16:33:17 web:2243] 200 GET /v1/metrics (10.212.1.18) 3.39ms
[I 210216 16:33:32 web:2243] 200 GET /v1/metrics (10.212.1.18) 1.37ms
[I 210216 16:33:42 cm_model:103] PROCESSING Feedback Event.
[I 210216 16:33:42 cm_model:104] {'Host': 'seldon-multiclass-model-metrics.production.svc.cluster.local:80', 'User-Agent': 'Go-http-client/1.1', 'Content-Length': '156', 'Ce-Endpoint': 'default', 'Ce-Id': 'e3f56826-ad3a-4638-9059-0d550c6186e9', 'Ce-Inferenceservicename': 'multiclass-model', 'Ce-Knativearrivaltime': '2021-02-16T16:33:41.195465611Z', 'Ce-Modelid': 'classifier', 'Ce-Namespace': 'production', 'Ce-Requestid': 'abc71989-f625-4ff4-bc9b-6b02967968d0', 'Ce-Source': 'http://:8000/', 'Ce-Specversion': '1.0', 'Ce-Time': '2021-02-16T16:33:40.82106025Z', 'Ce-Traceparent': '00-11bf7dd2281de401dfdb00f41b9c6cc7-89240c365d3cefb2-00', 'Ce-Type': 'io.seldon.serving.feedback', 'Content-Type': 'application/json', 'Traceparent': '00-11bf7dd2281de401dfdb00f41b9c6cc7-b017a64a85547fde-00', 'Accept-Encoding': 'gzip'}
[I 210216 16:33:42 cm_model:105] ----
[W 210216 16:33:42 base:269] GET http://elasticsearch-master.seldon-logs.svc.cluster.local:9200/inference-log-production-seldon-multiclass-model-default/_doc/abc71989-f625-4ff4-bc9b-6b02967968d0 [status:404 request:0.331s]
[E 210216 16:33:42 web:1793] Uncaught exception POST / (10.212.1.24)
    HTTPServerRequest(protocol='http', host='seldon-multiclass-model-metrics.production.svc.cluster.local:80', method='POST', uri='/', version='HTTP/1.1', remote_ip='10.212.1.24')
    Traceback (most recent call last):
      File "/opt/conda/lib/python3.7/site-packages/tornado/web.py", line 1702, in _execute
        result = method(*self.path_args, **self.path_kwargs)
      File "/microservice/adserver/server.py", line 242, in post
        response = self.model.process_event(request, headers)
      File "/microservice/adserver/cm_model.py", line 159, in process_event
        error, status_code=400, reason="METRICS_SERVER_ERROR"
    seldon_core.flask_utils.SeldonMicroserviceException
```

### PR DESCRIPTION
**Pull request by fg91**
**What this PR does / why we need it**:

The `elasticsearch_index = f"inference-log-{seldon_namespace}-seldon-...` of the metrics server is only correct if the namespace of the model happens to be `seldon` as the the elasticsearch index is called `'inference-log-seldon-<namespace>...'`

**Which issue(s) this PR fixes**:
Fixes #2971

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

### CODE REVIEW

**Comment by seldondev** [comment]
Hi @fg91. Thanks for your PR.

I'm waiting for a [SeldonIO](https://github.com/orgs/SeldonIO/people) or [todo](https://github.com/orgs/todo/people) member to verify that this patch is reasonable to test. If it is, they should reply with `/ok-to-test` on its own line. Until that is done, I will not automatically test new commits in this PR, but the usual testing commands by org members will still work. Regular contributors should [join the org](https://github.com/orgs/todo/people) to skip this step.

Once the patch is verified, the new status will be reflected by the `ok-to-test` label.

I understand the commands that are listed [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository.
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **cliveseldon**
You can assign the PR to them by writing `/assign @cliveseldon` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by ryandawsonuk** [comment]
@fg91 yes you're right, serving engine should come before namespace as it does in the request logger code
https://github.com/SeldonIO/seldon-core/blob/master/components/seldon-request-logger/app/log_helper.py#L44

And is mentioned in its [README](https://github.com/SeldonIO/seldon-core/tree/master/components/seldon-request-logger#local-testing-against-seldon-executor)

/lgtm
/approve

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ryandawsonuk](https://github.com/SeldonIO/seldon-core/pull/2972#issuecomment-780386327 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ryandawsonuk](https://github.com/SeldonIO/seldon-core/pull/2972#issuecomment-780386327 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ryandawsonuk — APPROVED** [review]

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *[ryandawsonuk](https://github.com/SeldonIO/seldon-core/pull/2972#pullrequestreview-591932627 "Approved")*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)~~ [ryandawsonuk]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
